e2e/testcafe-devextreme/tests/dataGrid/common/focus/focusedRow/focusedRow.ts

```diff
@@ -987,3 +987,56 @@ test('Group should expand when focusedRowKey is set and data items have \'items\
   height: 400,
   focusedRowEnabled: true,
 }));
+
+// T1292991
+test('Focused row index should be updated after updating a row when repaintChangesOnly is enabled', async (t) => {
```

Comment:
Suggestion: since the fix was quite complex with taking into account previous tickets and refactoring, can we add matrix tests that will cover the issue more widely?

Like: one loop for repaintChangesOnly: [true, false] and another loop for changeType: ['insert', 'remove', 'update']. And in the test we check that focusedRowKey and focusedRowIndex are correct and that focusedRowChanged, focusedRowChanging events are called.

Reply:
Also I suggest to rewrite the test to jest, because this test is fully functional